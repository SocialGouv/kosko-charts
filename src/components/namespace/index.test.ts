import { create, Params } from "./index";

test.each([
  ["because of missing variables", {} as Params],
  [
    "because the namespace name is emptys",
    {
      namespace: { name: "" },
    } as Params,
  ],
])("should throw %s", (_: string, params: Params) => {
  expect(() => create(params)).toThrowErrorMatchingSnapshot();
});

const validParams: Params = {
  domain: "",
  namespace: { name: "sample-42-my-test" },
  subdomain: "",
};
test.each([
  ["a namespace", {}],
  [
    "a namespace with extra labels and annotations",
    {
      annotations: {
        "app.gitlab.com/app": "socialgouv-sample",
        "app.gitlab.com/env": "my-test",
      },
      labels: {
        application: "sample",
        owner: "sample",
        team: "sample",
      },
      namespace: { name: "sample-42-my-test" },
    },
  ],
])("should return %s", (_: string, params: Partial<Params>) => {
  expect(create({ ...validParams, ...params })).toMatchSnapshot();
});
