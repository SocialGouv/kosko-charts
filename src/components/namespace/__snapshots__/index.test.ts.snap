// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should return a namespace 1`] = `
Object {
  "namespace": Object {
    "apiVersion": "v1",
    "kind": "Namespace",
    "metadata": Object {
      "labels": Object {
        "app": "sample-42-my-test",
      },
      "name": "sample-42-my-test",
    },
  },
}
`;

exports[`should return a namespace with extra labels and annotations 1`] = `
Object {
  "namespace": Object {
    "apiVersion": "v1",
    "kind": "Namespace",
    "metadata": Object {
      "annotations": Object {
        "app.gitlab.com/app": "socialgouv-sample",
        "app.gitlab.com/env": "my-test",
      },
      "labels": Object {
        "app": "sample-42-my-test",
        "application": "sample",
        "owner": "sample",
        "team": "sample",
      },
      "name": "sample-42-my-test",
    },
  },
}
`;

exports[`should throw because of missing variables 1`] = `
"BadArgument:required property \\"namespace\\"
└─ cannot decode undefined, should be object"
`;

exports[`should throw because the namespace name is emptys 1`] = `
"BadArgument:required property \\"namespace\\"
└─ required property \\"name\\"
   └─ cannot decode \\"\\", should be NonEmptyString"
`;
